test-apps/imodel-transformer/src/ElementUtils.ts

```diff
@@ -0,0 +1,74 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import { DbResult, Id64Set, Id64String } from "@bentley/bentleyjs-core";
+import {
+  Category, CategorySelector, DisplayStyle, ECSqlStatement, GeometricModel3d, IModelDb, ModelSelector, SubCategory,
+} from "@bentley/imodeljs-backend";
+
+export namespace ElementUtils {
+
+  function queryElementIds(iModelDb: IModelDb, classFullName: string): Id64Set {
+    const elementIds = new Set<Id64String>();
+    iModelDb.withPreparedStatement(`SELECT ECInstanceId FROM ${classFullName}`, (statement: ECSqlStatement) => {

```

Comment:
might there be advantages with preferring `iModelDb.query` which returns an async iterator? (e.g. prevent blocking)

Reply:
*Validation* is really only relevant for finding issues or confirming fixes, so I think it might be better to ask the same question of the IModelExporter and IModelTransformer core. That is, would making additional things async have a positive or negative impact on performance.

Reply:
So let's consider it out of scope of this PR but worth visiting for performance in the future, and I'll resolve.